vendor/k8s.io/kubernetes/pkg/kubectl/resource_printer_test.go

```diff
@@ -241,9 +263,44 @@ func TestCustomTypePrinting(t *testing.T) {
 	}
 }
 
+func TestCustomTypePrintingWithKind(t *testing.T) {
+	columns := []string{"Data"}
+	printer := NewHumanReadablePrinter(&PrintOptions{WithKind: true, Kind: "test"})
+	printer.Handler(columns, PrintCustomType)
+
+	obj := TestPrintType{"test object"}
+	buffer := &bytes.Buffer{}
+	err := printer.PrintObj(&obj, buffer)
+	if err != nil {
+		t.Fatalf("An error occurred printing the custom type: %#v", err)
+	}
+	expectedOutput := "Data\ntest/test object"
+	if buffer.String() != expectedOutput {
+		t.Errorf("The data was not printed as expected. Expected:\n%s\nGot:\n%s", expectedOutput, buffer.String())
+	}
+}
+
+func TestCustomTypePrintingWithForcedKind(t *testing.T) {
+	columns := []string{"Data"}
+	printer := NewHumanReadablePrinter(&PrintOptions{WithKind: false})
+	printer.EnsurePrintWithKind("test")
```

Comment:
Ah, I am not sure if this can be truly approached from here, as it is determined in [kubernetes/pkg/kubectl/cmd/get.go](https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/kubectl/cmd/get.go#L307) whether or not kind will be included. Another approach to testing this could be to make a test in `hack/test-cmd.sh` and expect an output of multiple known resources to be formatted as `kind/name`